src/Aspire.Dashboard/ConsoleLogs/LogEntries.cs

```diff
@@ -65,15 +67,23 @@ void InsertLogEntry(List<LogEntry> logEntries, int index, LogEntry logEntry)
             }
             else
             {
-                logEntry.LineNumber = logEntries[index - 1].LineNumber + 1;
+                logEntry.LineNumber = _logEntries[index - 1].LineNumber + 1;
+            }
+
+            // Trim old log messages if we have a maximum and we're over it.
+            if (MaximumEntryCount is not (null or 0) && _logEntries.Count >= MaximumEntryCount && index is not 0)
+            {
+                _logEntries.RemoveAt(0);
```

Comment:
Removing at index 0 will copy all the data of the list back one position. At 10,000 items, and possibly hundreds of log entries a second this could consume a lot of machine resources.

I wrote `CircularBuffer<T>` for telemetry that handles this automatically (removing oldest just involves changing start/end indexes).

I think you should see whether `CircularBuffer<T>` makes sense here.

Reply:
I'd considered adding a circular buffer here but didn't realise we already had one in the code. Will use it.